packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1226,6 +1260,16 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
       },
     );
 
+    final int? semanticsMaxValueLength;
+    if (widget.maxLengthEnforced &&
+      _effectiveMaxLengthEnforcement != MaxLengthEnforcement.none &&
+      widget.maxLength != null &&
+      widget.maxLength! > 0) {
+      semanticsMaxValueLength = widget.maxLength;
+    } else {
+      semanticsMaxValueLength = null;
```

Comment:
NIt: You can get rid of this `else` block and it will still be null anyway.

Reply:
Actually it can't. Because without this control flow the `final xxx` without assignment is a compile error.

Reply:
Ah NNBD is still tripping me up.  Good point. 👍